torch/csrc/distributed/c10d/intra_node_comm.cpp

```diff
@@ -35,8 +40,34 @@ static NvlMesh getNvlMesh(const std::vector<int>& rankToDeviceIdx) {
     }
   }
   return nvlMesh;
+#else
+  NvlMesh nvlMesh = {};
```

Comment:
FYI I've moved the p2p connectivity detection logic to https://github.com/pytorch/pytorch/blob/cfdf658a919506c20805b9d241bb3b765227955f/torch/csrc/distributed/c10d/CudaDMAConnectivity.cpp#L4

It would be great if we can move these as well.